src/main/java/org/folio/marccat/resources/FixedFieldCodesGroupAPI.java

```diff
@@ -1,34 +1,47 @@
+import static org.folio.marccat.resources.shared.FixedFieldUtils.injectAuthorityLeaderCodes;
+import static org.folio.marccat.resources.shared.FixedFieldUtils.injectAuthorityMaterialCodes;
+import static org.folio.marccat.resources.shared.FixedFieldUtils.injectDefaultValues;
+import static org.folio.marccat.resources.shared.FixedFieldUtils.injectLeaderCodes;
+import static org.folio.marccat.resources.shared.FixedFieldUtils.injectMaterialCodes;
+import static org.folio.marccat.resources.shared.FixedFieldUtils.injectPhysicalDescriptionCodes;
 
 import java.util.HashMap;
 import java.util.Map;
 
-import static java.util.Optional.ofNullable;
-import static org.folio.marccat.config.constants.Global.*;
-import static org.folio.marccat.config.log.Message.MOD_MARCCAT_00017_CODES_GROUPS_NOT_AVAILABLE;
-import static org.folio.marccat.integration.MarccatHelper.doGet;
-import static org.folio.marccat.resources.shared.FixedFieldUtils.*;
+import org.folio.marccat.config.constants.Global;
+import org.folio.marccat.integration.AuthorityStorageService;
+import org.folio.marccat.resources.domain.FixedFieldCodesGroup;
+import org.folio.marccat.shared.CatalogingInformation;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 @RequestMapping(value = BASE_URI, produces = "application/json")
 public class FixedFieldCodesGroupAPI extends BaseResource implements CatalogingInformation {
 
   @GetMapping("/fixed-fields-code-groups-by-leader")
-  public FixedFieldCodesGroup getFixedFieldCodesGroupsByLeader(
-    @RequestParam final String leader,
-    @RequestParam(name = "code", defaultValue = Global.MATERIAL_TAG_CODE) final String code,
-    @RequestParam final String lang,
-    @RequestParam(required = false) final String valueField,
-    @RequestHeader(OKAPI_TENANT_HEADER_NAME) final String tenant,
-    @RequestHeader(Global.OKAPI_URL) String okapiUrl) {
+  public FixedFieldCodesGroup getFixedFieldCodesGroupsByLeader(@RequestParam final String leader,
+      @RequestParam(name = "code", defaultValue = Global.MATERIAL_TAG_CODE) final String code,
+      @RequestParam final String lang, @RequestParam(required = false) final String valueField,
+      @RequestHeader(OKAPI_TENANT_HEADER_NAME) final String tenant, @RequestHeader(Global.OKAPI_URL) String okapiUrl) {
     return doGet((storageService, configuration) -> {
       final FixedFieldCodesGroup fixedFieldCodesGroup = new FixedFieldCodesGroup();
-      final Map<String, Object> map = storageService.getMaterialTypeInfosByLeaderValues(leader.charAt(6), leader.charAt(7), code);
+      final Map<String, Object> map = storageService.getMaterialTypeInfosByLeaderValues(leader.charAt(6),
```

Comment:
Is this safe?  Do we know that `leader` is long enough, or could this result in an IndexOutOfBoundsException?

Reply:
It is bibliographic, @carmentrazza could you answer him, please?

Reply:
The leader is 24 characters long, there should be no problem.